ably/rest_channel.go

```diff
@@ -59,6 +62,37 @@ func (c *RestChannel) PublishAll(messages []*proto.Message) error {
 			v.ChannelOptions = c.options
 		}
 	}
+	useIdempotent := !c.client.opts.NoIdempotentRestPublishing
+	if useIdempotent {
+		switch len(messages) {
+		case 1:
+			// spec RSL1k2 we preserve the id if we have one message and it contains the
+			// id.
+			if messages[0].ID == "" {
```

Comment:
I assumed that if there was no `id` present in a message then it would be null, and not the empty string. The empty string isn't a valid id.

Also, I'd not noticed before, but why `ID` and not `Id` ? I realise it needs to have the initial capital, but the field is declared in the spec as `id`, and we should be making the minimum change to the spec to meet the language requirements. Similarly we have `ClientID` instead of `ClientId` etc.

Reply:
strings can not be `nil` so lack of id is an empty string. only pointers or interfaces can be nil(`null`).